test/ConfigManagementHelper.js

```diff
@@ -0,0 +1,715 @@
+    cmd = `${cmd} --app ${appIdentifier}`;
+  }
+
+  passConfigFileToCli(cmd, config, filePath, done);
+};
+
+env.exportEnv = function (envIdentifier, appIdentifier, done) {
+  const fileName = `${TestsHelper.randomStrings.plainString(10)}.json`;
+  const filePath = path.join(TestsHelper.ConfigFilesDir, fileName);
+  let cmd = `env export ${filePath} --output json`;
+  if (envIdentifier) {
+    cmd = `${cmd} ${envIdentifier}`;
+  }
+
+  if (appIdentifier) {
+    cmd = `${cmd} --app ${appIdentifier}`;
+  }
+
+  TestsHelper.execCmdWoMocks(cmd, null, (err) => {
+    if (err) {
+      return done(err);
+    }
+
+    fs.readFile(filePath, null, (err, data) => {
+      if (err) {
+        return done(err);
+      }
+
+      done(null, JSON.parse(data));
+    });
+  });
+};
+
+env.assertEnvOnly = function assertEnvOnly(envFromConfig, envId, envName, done) {
+  ApiService.envs.get(envId, (err, actualEnv) => {
+    if (err) {
+      return done(err);
+    }
+
+    expect(actualEnv.name).to.equal(envName);
+
+    const envSettings = envFromConfig.settings;
+    if (envSettings && envSettings.emailVerification) {
+      expect(actualEnv.emailVerification).to.deep.equal(envSettings.emailVerification);
+    } else {
+      expect(actualEnv).to.not.have.property('emailVerification');
+    }
+
+    let expectedApiVersion = 3;
+    if (envSettings && envSettings.apiVersion) {
+      expectedApiVersion = envSettings.apiVersion;
+    }
+
+    expect(actualEnv.apiVersion).to.equal(expectedApiVersion);
+
+    done();
+  });
+};
+
+env.buildExpectedPermissionsPerColl = function buildExpectedPermissionsPerColl(collList, rolesNameIdPairs) {
+  const expectedPermissionsPerColl = {};
+
+  collList.forEach((coll) => {
+    let expectedPermissions;
+    const collPermissionsInConfig = coll.permissions;
+    if (typeof collPermissionsInConfig === 'string') {
+      expectedPermissions = BackendCollectionPermission[collPermissionsInConfig];
+    } else {
+      expectedPermissions = {};
+      const rolesNames = Object.keys(collPermissionsInConfig);
+      rolesNames.forEach((roleName) => {
+        const permissionsPerRole = Object.keys(collPermissionsInConfig[roleName]);
+        permissionsPerRole.forEach((permission) => { // e.g. create, update
+          if (!expectedPermissions[permission]) {
+            expectedPermissions[permission] = [];
+          }
+
+          expectedPermissions[permission].push({
+            roleId: rolesNameIdPairs[roleName] || 'all-users',
+            type: collPermissionsInConfig[roleName][permission]
+          });
+        });
+      });
+    }
+
+    expectedPermissionsPerColl[coll.name] = expectedPermissions;
+  });
+
+  return expectedPermissionsPerColl;
+};
+
+env.assertCollections = function assertCollections(envId, collList, expCollCount, rolesNameIdPairs, done) {
+  ApiService.colls.get(envId, null, (err, actualColls) => {
+    if (err) {
+      return done(err);
+    }
+
+    expect(expCollCount).to.equal(actualColls.length);
```

Comment:
'actual' and 'expected' places should be exchanged. There are several places in this file with this problem - can be found with search by 'expect(exp'.